### ISSUE TITLE
Add screen reader announcements for validation errors

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-text-field#117]
For example, when a maxlength requirement is reached, the screen reader should announce that, or some other audible cue should be given. Not sure if any screen readers do this already, but we should make sure.

**Comment by juuso-vaadin** [comment]
Currently the validation happens on `blur` event so focus is already on the next element when the validation error appears. Would it be possible to allow customizing the validation trigger? For example one could set validations to run 2 seconds after last keystroke.

**Comment by juuso-vaadin** [comment]
Also related to issue: https://github.com/vaadin/vaadin-text-field/issues/16

Validation appears on `blur` event so screen readers are not able to catch the changing state. Would it be possible to allow developer to change the validation to be triggered with a timeout from last keystoke for example?

**Comment by rolfsmeds** [comment]
I'd say it's fine that validation only happens once the user moves onward from the field, and thus has to move focus back to it to correct the error, but the validation error should definitely be announced immediately once validation happens, regardless of where the user's focus is.
